testsuite/MDAnalysisTests/converters/test_openmm_parser.py

```diff
@@ -130,6 +155,72 @@ class TestOpenMMTopologyParser(OpenMMTopologyBase):
     expected_n_bonds = 1922
 
 
+class TestOpenMMTopologyParserWithNoElements(OpenMMTopologyBase):
+    ref_filename = app.PDBFile(PDB).topology
+    expected_n_atoms = 47681
+    expected_n_residues = 11302
+    expected_n_segments = 1
+    expected_n_bonds = 25533
+
+    expected_attrs = [
+        "ids",
+        "names",
+        "resids",
+        "resnames",
+        "masses",
+        "bonds",
+        "chainIDs",
+    ]
+
+    @pytest.fixture()
+    def top(self, filename):
+        omm_topology = self.ref_filename
+        for a in omm_topology.atoms():
+            a.element = None
+        return self.parser(PDB) \
+                   ._mda_topology_from_omm_topology(omm_topology)
+
+    def test_no_elements(self, top):
+        omm_topology = self.ref_filename
+        for a in omm_topology.atoms():
+            a.element = None
+
+        wmsg = ("Element information is missing, elements attribute "
+                "will not be populated. "
+                "Atomtype attribute will be guessed using atom "
+                "name and mass will be guessed using atomtype."
+                "See MDAnalysis.topology.guessers."
+                )
+        with pytest.warns(UserWarning, match=wmsg):
+            mda_top = self.parser(PDB) \
+                      ._mda_topology_from_omm_topology(omm_topology)
```

Comment:
I don't think this is necessary either, you should be able to directly parse the modified topology (proposed in my previous comment). Also the name of this test should be better. Would this work?

```suggestion
    def test_no_elements_warn(self, top):
        wmsg = ("Element information is missing, elements attribute "
                "will not be populated. "
                "Atomtype attribute will be guessed using atom "
                "name and mass will be guessed using atomtype."
                "See MDAnalysis.topology.guessers.")

        with pytest.warns(UserWarning, match=wmsg):
            self.parser(self.filename)
```

Reply:
Calling the parser that way doesn't raise any warnings for testing so I've replaced the call with a more workable one